pkg/minikube/notify/notify.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notify
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/golang/glog"
+	"github.com/spf13/viper"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/version"
+)
+
+const updateLinkPrefix = "https://github.com/kubernetes/minikube/releases/tag/v"
+
+var (
+	timeLayout                = time.RFC1123
+	lastUpdateCheckFilePath   = constants.MakeMiniPath("last_update_check")
+	githubMinikubeReleasesURL = "https://api.github.com/repos/kubernetes/minikube/releases"
+)
+
+func MaybePrintUpdateTextFromGithub(outputFile *os.File) {
+	MaybePrintUpdateText(outputFile, githubMinikubeReleasesURL, lastUpdateCheckFilePath)
+}
+
+func MaybePrintUpdateText(outputFile *os.File, url string, lastUpdatePath string) {
```

Comment:
This should probably just be "output", and you can just make it an io.Writer instead of a file. That should answer your question about the test as well (from line 132).